src/main/java/dev/plex/extras/command/EnchantCommand.java

```diff
@@ -36,27 +35,72 @@ protected Component execute(@NotNull CommandSender sender, @Nullable Player play
         ItemStack item = player.getInventory().getItemInMainHand();
         if (item.getType() == Material.AIR)
         {
-            return MiniMessage.miniMessage().deserialize("<red>You must be holding an enchantable item.");
+            return messageComponent("enchantMustHoldItem");
         }
 
-        if (args.length == 1)
+        switch (args[0].toLowerCase())
         {
-            if (args[0].equalsIgnoreCase("list"))
-            {
-                return MiniMessage.miniMessage().deserialize("<dark_gray>All possible enchantments are for this item are: <gray>" + StringUtils.join(getEnchantmentNames(item), ", "));
-            }
-            if (args[0].equalsIgnoreCase("addall"))
-            {
+            case "add":
+                if (args.length < 2)
+                {
+                    return messageComponent("enchantSpecify");
+                }
+
+                Enchantment enchantmentToAdd = Registry.ENCHANTMENT.get(NamespacedKey.minecraft(args[1].toLowerCase()));
+                if (enchantmentToAdd == null || !enchantmentToAdd.canEnchantItem(item))
+                {
+                    return messageComponent("enchantInvalid");
+                }
+
+                int levelToAdd = enchantmentToAdd.getMaxLevel();
+                if (args.length >= 3)
+                {
+                    try
+                    {
+                        levelToAdd = Integer.parseInt(args[2]);
+                        if (levelToAdd < 1 || levelToAdd > 255)
+                        {
+                            return messageComponent("enchantInvalidLevel");
+                        }
+                    }
+                    catch (NumberFormatException e)
+                    {
+                        return messageComponent("enchantInvalidLevel");
+                    }
+                }
+
+                item.addUnsafeEnchantment(enchantmentToAdd, levelToAdd);
+                player.playSound(player.getLocation(), Sound.BLOCK_ANVIL_USE, 1, 1);
+                return messageComponent("enchantAdd", enchantmentToAdd.getKey().getKey(), levelToAdd);
```

Comment:
Wouldn't getKey() return an non-capitalized name of the enchantment?

Reply:
https://www.grammarly.com/blog/capitalization-rules/

Reply:
@VideoGameSmash12 no, it doesn't